HIP/hip-1259.md

```diff
@@ -0,0 +1,167 @@
+staking rewards apply during a transaction, only 0.0.800 updates—still lightweight due to its simple structure.
+
+This design builds on basic principles: batch operations for efficiency, single-point collection for simplicity, and
+delayed distributions to align with existing staking cycles.
+
+# User Stories
+
+1. As a user, I want transaction records to show a simple fee payment, without a tangle of transfers to multiple system
+   accounts.
+2. As a block node or mirror node operator, I want smaller block streams to lower my data ingestion and storage costs.
+3. As a network administrator, I want fee handling that maintains incentives for nodes while improving overall
+   throughput.
+
+# Specification
+
+### Overview
+
+Create a new special account at address 0.0.802 (the actual number is configurable, like 0.0.800 and 0.0.801). This
+account has no keys, does not accept HBAR transfers, and is controlled solely by network code—preventing token
+associations or updates of any kind.
+
+Add a singleton state in the Merkle tree called `NodePaymentsInfo`. This is an map where each key is a node `AccountID`
+number, and each value is a node receiving node fees.
+
+During transaction handling:
+
+- Calculate fees as usual.
+- Add the node fee to the corresponding entry in the `NodePaymentsInfo` map.
+- Transfer *all* fees into 0.0.802 via a single entry in the transaction's transfer list.
+
+![Fee Collection and Disbursement Flow](../assets/hip-1259/fee-collection-account-flow.png "Fee Collection and Disbursement Flow")
+
+The singleton updates the Merkle tree per transaction, but records in the block stream only once per block. Since
+0.0.802 has no keys, its block stream record is minimal.
+
+At the end of each staking period, prior to any other transactions or operations:
+
+1. Create a synthetic transaction to distribute all hbars from 0.0.802. This single transaction can have an a large
+   transfer list accommodating the transfers to all nodes and the special accounts
+2. Pay node fees from the `NodePaymentsInfo` map to each node's account.
+    - If for any reason the node’s account cannot accept the fees (i.e. it is deleted or doesn’t exist), then they are
+      forfeit. They will remain in 0.0.802 until step 3
+    - `receiverSigRequired` is ignored for payments to node accounts
+    - Hooks are not executed when fees are paid to node accounts
+3. Split the remaining balance among 0.0.98, 0.0.800, and 0.0.801 per current rules
+4. Reset `NodePaymentsInfo` to an empty map.
+
+This synthetic transaction is recorded in the block stream like any other.
+
+### Protobuf Changes
+
+The singleton state is defined in a new protobuf file: `services/state/node_payments.proto`.
+
+```protobuf
+// A singleton state object that accumulates node fees for distribution.
+message NodePaymentsInfo {
+  // A map from account_number to NodePaymentInfo.
+  map<int64, NodePaymentInfo> payments = 1;
```

Comment:
The info suffix seems unnecessary and makes it harder to say when plural. Could consider dropping it unless it's some standard naming for state objects.

```suggestion
message NodePayments {
  // A map from account_number to NodePaymentInfo.
  map<int64, NodePayment> payments = 1;
```

Reply:
Applied